tests/symbolicator/test_payload_full.py

```diff
@@ -185,3 +187,82 @@ def test_missing_debug_images(self):
         event = self.post_and_retrieve_event(payload)
         assert event.data["culprit"] == "unknown"
         insta_snapshot_stacktrace_data(self, event.data)
+
+    def test_resolving_with_candidates_sentry_source(self):
+        # Checks the candidates with a sentry source URI for location
+        file = File.objects.create(
+            name="crash.pdb", type="default", headers={"Content-Type": "text/x-breakpad"}
+        )
+
+        path = get_fixture_path("windows.sym")
+        with open(path, "rb") as f:
+            file.putfile(f)
+
+        ProjectDebugFile.objects.create(
+            file=file,
+            object_name="crash.pdb",
+            cpu_name="x86",
+            project=self.project,
+            debug_id="3249d99d-0c40-4931-8610-f4e4fb0b6936-1",
+            code_id="5AB380779000",
+        )
+
+        self.login_as(user=self.user)
+
+        event_data = {
+            "contexts": {
+                "device": {"arch": "x86"},
+                "os": {"build": "", "name": "Windows", "type": "os", "version": "10.0.14393"},
+            },
+            "debug_meta": {
+                "images": [
+                    {
+                        "id": "3249d99d-0c40-4931-8610-f4e4fb0b6936-1",
+                        "image_addr": "0x2a0000",
+                        "image_size": 36864,
+                        "name": "C:\\projects\\breakpad-tools\\windows\\Release\\crash.exe",
+                        "type": "symbolic",
+                    }
+                ]
+            },
+            "exception": {
+                "stacktrace": {
+                    "frames": [
+                        {
+                            "function": "<unknown>",
+                            "instruction_addr": "0x2a2a3d",
+                            "package": "C:\\projects\\breakpad-tools\\windows\\Release\\crash.exe",
+                        }
+                    ]
+                },
+                "thread_id": 1636,
+                "type": "EXCEPTION_ACCESS_VIOLATION_WRITE",
+                "value": "Fatal Error: EXCEPTION_ACCESS_VIOLATION_WRITE",
+            },
+            "platform": "native",
+            "timestamp": iso_format(before_now(seconds=1)),
+        }
+
+        with self.feature("organizations:images-loaded-v2"):
+            event = self.post_and_retrieve_event(event_data)
+        assert event.data["culprit"] == "main"
+
+        candidates = event.data["debug_meta"]["images"][0]["candidates"]
+        redact_location(candidates)
+        self.insta_snapshot(candidates)
```

Comment:
You could also inspect the candidates list directly and check if the location is `f"sentry://project_debug_file/{debug_file.id}"`. You can get the debug file in the return value from `ProjectDebugFile.objects.create`.

Reply:
I think I still want the insta snapshot for the other fields i think.  Not sure if the specific ID is important enough to test, let's not for now as that's less brittle, until this is known to be a problem.